collect_app/src/main/java/org/odk/collect/android/map/GoogleMapFragment.java

```diff
@@ -197,23 +201,45 @@ protected LatLngBounds expandBounds(LatLngBounds bounds, double factor) {
         return new LatLngBounds(new LatLng(south, west), new LatLng(north, east));
     }
 
+    protected void moveOrAnimateCamera(CameraUpdate movement, boolean animate) {
+        if (animate) {
+            map.animateCamera(movement);
+        } else {
+            map.moveCamera(movement);
+        }
+    }
+
+    @Override public int addMarker(MapPoint point, boolean draggable) {
+        int featureId = nextFeatureId++;
+        features.put(featureId, new MarkerFeature(map, point, draggable));
+        return featureId;
+    }
+
+    @Override public @Nullable MapPoint getMarkerPoint(int featureId) {
+        MapFeature feature = features.get(featureId);
+        if (feature instanceof MarkerFeature) {
```

Comment:
Don't you like ternary operators? I would prefer it here.

Reply:
Sure!  Sometimes I use them; in this case it didn't occur to me.  Fixed.